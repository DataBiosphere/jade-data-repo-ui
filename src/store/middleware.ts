import createSagaMiddleware from 'redux-saga';

export const sagaMiddleware = createSagaMiddleware();

const middleware = [sagaMiddleware];

/* istanbul ignore next */
if (process.env.NODE_ENV === 'development') {
  //eslint-disable-next-line  @typescript-eslint/no-var-requires
  const { createLogger } = require('redux-logger');
  //eslint-disable-next-line  @typescript-eslint/no-var-requires
  const invariant = require('redux-immutable-state-invariant').default;

  middleware.push(invariant());
  middleware.push(createLogger({ collapsed: true }));
}

export default middleware;
