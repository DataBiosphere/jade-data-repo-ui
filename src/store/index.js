import { applyMiddleware, createStore, compose, combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';

import rootSaga from 'sagas/index';
import rootReducer from 'reducers/index';
import history from 'modules/hist';
import middleware, { sagaMiddleware } from './middleware';

const reducer = (hist) =>
  combineReducers({
    router: connectRouter(hist),
    ...rootReducer,
  });

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/* istanbul ignore next */
const configStore = (initialState = {}) => {
  const store = createStore(
    reducer(history),
    initialState,
    composeEnhancer(applyMiddleware(...middleware)),
  );

  sagaMiddleware.run(rootSaga);

  if (module.hot) {
    module.hot.accept('reducers', () => {
      store.replaceReducer(require('reducers/index').default);
    });
  }

  return {
    store,
  };
};

const { store } = configStore();

global.store = store;

export { store };
