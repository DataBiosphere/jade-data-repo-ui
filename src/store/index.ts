import { applyMiddleware, combineReducers, compose, createStore, Store } from 'redux';
import { connectRouter } from 'connected-react-router';

import rootSaga from 'sagas/index';
import rootReducer, { initialTdrState, TdrState } from 'reducers/index';
import history from 'modules/hist';
import { History } from 'history';
import middleware, { sagaMiddleware } from './middleware';

const reducer = (hist: History) =>
  combineReducers({
    router: connectRouter(hist),
    ...rootReducer,
  });

// @ts-ignore
const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/* istanbul ignore next */
const configStore = (initialState: TdrState = initialTdrState): Store<TdrState> => {
  const store = createStore(
    reducer(history),
    initialState,
    composeEnhancer(applyMiddleware(...middleware)),
  );

  sagaMiddleware.run(rootSaga);

  // @ts-ignore
  if (module.hot) {
    // @ts-ignore
    module.hot.accept('reducers', () => {
      store.replaceReducer(require('reducers/index').default);
    });
  }

  return store;
};

const store = configStore();

// @ts-ignore
global.store = store;

export { store };
