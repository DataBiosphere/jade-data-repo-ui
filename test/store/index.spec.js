import { store } from 'store';

describe('store', () => {
  it('should have a store', () => {
    expect(store.getState()).toMatchSnapshot();
  });
});
