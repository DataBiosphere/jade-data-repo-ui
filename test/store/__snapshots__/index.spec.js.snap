// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store should have a store 1`] = `
Object {
  "app": Object {
    "alerts": Array [],
  },
  "dataset": Object {
    "description": "",
    "name": "",
    "readers": Array [],
    "study": "",
  },
  "forms": Object {
    "$form": Object {
      "errors": Object {},
      "focus": false,
      "initialValue": Object {
        "dataset": Object {
          "description": "",
          "name": "",
          "readers": Array [],
          "study": "",
        },
      },
      "intents": Array [],
      "model": "",
      "pending": false,
      "pristine": true,
      "retouched": false,
      "submitFailed": false,
      "submitted": false,
      "touched": false,
      "valid": true,
      "validated": false,
      "validating": false,
      "validity": Object {},
      "value": Object {
        "dataset": Object {
          "description": "",
          "name": "",
          "readers": Array [],
          "study": "",
        },
      },
    },
    "dataset": Object {
      "$form": Object {
        "errors": Object {},
        "focus": false,
        "initialValue": Object {
          "description": "",
          "name": "",
          "readers": Array [],
          "study": "",
        },
        "intents": Array [],
        "model": "dataset",
        "pending": false,
        "pristine": true,
        "retouched": false,
        "submitFailed": false,
        "submitted": false,
        "touched": false,
        "valid": true,
        "validated": false,
        "validating": false,
        "validity": Object {},
        "value": Object {
          "description": "",
          "name": "",
          "readers": Array [],
          "study": "",
        },
      },
      "description": Object {
        "errors": Object {},
        "focus": false,
        "initialValue": "",
        "intents": Array [],
        "model": "dataset.description",
        "pending": false,
        "pristine": true,
        "retouched": false,
        "submitFailed": false,
        "submitted": false,
        "touched": false,
        "valid": true,
        "validated": false,
        "validating": false,
        "validity": Object {},
        "value": "",
      },
      "name": Object {
        "errors": Object {},
        "focus": false,
        "initialValue": "",
        "intents": Array [],
        "model": "dataset.name",
        "pending": false,
        "pristine": true,
        "retouched": false,
        "submitFailed": false,
        "submitted": false,
        "touched": false,
        "valid": true,
        "validated": false,
        "validating": false,
        "validity": Object {},
        "value": "",
      },
      "readers": Object {
        "$form": Object {
          "errors": Object {},
          "focus": false,
          "initialValue": Array [],
          "intents": Array [],
          "model": "dataset.readers",
          "pending": false,
          "pristine": true,
          "retouched": false,
          "submitFailed": false,
          "submitted": false,
          "touched": false,
          "valid": true,
          "validated": false,
          "validating": false,
          "validity": Object {},
          "value": Array [],
        },
      },
      "study": Object {
        "errors": Object {},
        "focus": false,
        "initialValue": "",
        "intents": Array [],
        "model": "dataset.study",
        "pending": false,
        "pristine": true,
        "retouched": false,
        "submitFailed": false,
        "submitted": false,
        "touched": false,
        "valid": true,
        "validated": false,
        "validating": false,
        "validity": Object {},
        "value": "",
      },
    },
  },
  "github": Object {
    "repos": Object {
      "data": Object {},
      "message": "",
      "query": "",
      "status": "idle",
    },
  },
  "user": Object {
    "isAuthenticated": false,
    "status": "idle",
  },
}
`;
